cmd/fpga_plugin/fpga_plugin.go

```diff
@@ -65,6 +65,16 @@ func parseMode(input string) pluginMode {
 	return wrongMode
 }
 
+func (mode pluginMode) String() string {
+	switch mode {
+	case afMode:
+		return "af"
```

Comment:
Hmm... "af" and "region" are used in more than one place. If string constants are inevitable then having just
```
type pluginMode string

const (
    afMode pluginMode = "af"
    regionMode pluginMode = "region"
)

func isValidPluginMode(m pluginMode) bool {
    if m == afMode || m == regionMode {
        return true
    }

    return false
}

func main() {
    ...
    mode := pluginMode(userInput)
    if !isValidPluginMode(mode) {
        glog.Error("Wrong mode")
        os.Exit(1)
    }
}
```
doesn't look too bad after all. At least we'd be able to catch type mismatches in assignments and comparisons.